src/components/ContactsApp.jsx

```diff
@@ -1,220 +0,0 @@
-import { Main, Content } from 'cozy-ui/react/Layout'
-import { withContacts, withContactsMutations } from '../connections/allContacts'
-import { getFullContactName } from '../helpers/contacts'
-import ContactImportationModal from './ContactImportationModal'
-
-const ContactsHeaderWithActions = ({ displayContactForm }, { t }) => (
-  <ContactsHeader
-    renderActions={() => {
-      const fakeintent = new URL(window.location).searchParams.get('fakeintent')
-      return (
-        <div className="actions">
-          {fakeintent !== null && (
-            <ContactsIntentButton
-              label={'Select a Contact (intent)'}
-              action="PICK"
-            />
-          )}
-          {fakeintent !== null && (
-            <ContactsIntentButton
-              label={'Create a Contact (intent)'}
-              action="CREATE"
-              data={{ me: true }}
-            />
-          )}
-          <OpenContactFormButton
-            onClick={displayContactForm}
-            label={t('create_contact')}
-          />
-        </div>
-      )
-    }}
-  />
-)
-ContactsHeaderWithActions.propTypes = {
-  displayContactForm: PropTypes.func.isRequired
-}
-
-const SelectionBarWithActions = ({
-  selected,
-  hideSelectionBar,
-  trashAction
-}) => (
-  <SelectionBar
-    selected={selected}
-    hideSelectionBar={hideSelectionBar}
-    actions={{
-      trash: {
-        action: trashAction
-      }
-    }}
-  />
-)
-SelectionBarWithActions.propTypes = {
-  selected: PropTypes.array.isRequired,
-  hideSelectionBar: PropTypes.func.isRequired,
-  trashAction: PropTypes.func.isRequired
-}
-
-class ContactsApp extends React.Component {
-  state = {
-    displayedContact: null,
-    isImportationDisplayed: false,
-    isCreationFormDisplayed: false
-  }
-
-  displayImportation = () => {
-    this.setState({
-      isImportationDisplayed: true
-    })
-  }
-
-  hideImportation = () => {
-    this.setState({
-      isImportationDisplayed: false
-    })
-  }
-
-  displayContactCard = contact => {
-    this.setState({
-      displayedContact: contact
-    })
-  }
-
-  onDeleteContact = contact => {
-    this.hideContactCard()
-    Alerter.info('delete-confirmation.deleted', {
-      name: getFullContactName(contact.name)
-    })
-  }
-
-  hideContactCard = () => {
-    this.setState({
-      displayedContact: null
-    })
-  }
-
-  displayContactForm = () => {
-    this.setState({
-      isCreationFormDisplayed: true
-    })
-  }
-
-  hideContactForm = () => {
-    this.setState({
-      isCreationFormDisplayed: false
-    })
-  }
-
-  onCreateContact = contact => {
-    this.hideContactForm()
-    this.displayContactCard(contact)
-  }
-
-  deleteSelectedContacts = async () => {
-    const { selection } = this.props
-    const promises = selection.map(contact => this.props.deleteContact(contact))
-    await Promise.all(promises)
-    this.props.clearSelection()
-  }
-
-  componentWillReceiveProps(nextProps) {
```

Comment:
I removed it as I could not find a way to manage it. Its lack is non blocking. That is a small non technical debt that we'll manage later.

Currently there is no global state management. So we are using an architecture with a main component that manages the global state, slices of this state are passed as props to underlying specific components. This architecture is just as respectable as a Flux architecture.

Nevertheless `componentWillReceiveProps` that implements a state management is a smell.

> Using this lifecycle method often leads to bugs and inconsistencies, and for that reason it is going to be deprecated in the future. — [componentWillReceiveProps()](https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops)

If the change breaks something, it will be easier to take the smell into consideration and find a way to add the missing feature (update the modal if I understood well).
Otherwise the bug will become a feature.

Reply:
I'm not quite sure I understand — not porting this piece of code will almost definitely produce a buggy behavior and cause a regression, so I don't see how it would be non blocking?

I'm aware of the current architecture and that we should remove the `componentWillReceiveProps` hook, but as far as I know using `componentDidUpdate` for this sort of work is perfectly fine and should be rather straightforward?

Reply:
If you say so. I guess you are up-to-date with the react documentation about [componentDidUpdate](https://reactjs.org/docs/react-component.html#componentdidupdate) lifecycle function:

> Use this as an opportunity to operate on the DOM when the component has been updated. This is also a good place to do network requests as long as you compare the current props to previous props (e.g. a network request may not be necessary if the props have not changed).

Once again, this is a smell that the architecture is not following the fundamental rules of react and its components lifecycle.

The bug that does not display something is not blocking as in "oops, reboot please".

A commit to ~~update~~ revert that is on its way.

Reply:
@y-lohse I did not check the behavior of the application after applying https://github.com/cozy/cozy-contacts/pull/215/commits/47652761604c1e9694803513a752dc416d33218c

But it breaks the application without sending any error message as it goes into a loop of render/componentDidUpdate/render…

![peek 2018-08-13 15-13](https://user-images.githubusercontent.com/701648/44036142-6a032212-9f11-11e8-93b2-e30209d81708.gif)